pkg/reloader/reloader.go

```diff
@@ -110,25 +115,43 @@ func (r *Reloader) Run(ctx context.Context) error {
 			if event.Op != fsnotify.Write && event.Op != fsnotify.Create {
 				continue
 			}
-
-			h := sha256.New()
-			f, err := os.Open(r.ConfigFile)
+			touchedInfo, err := os.Stat(event.Name)
 			if err != nil {
-				log.Printf("Error: %s\n", err)
-				continue
-			}
-			if _, err := io.Copy(h, f); err != nil {
-				log.Printf("Error: %s\n", err)
 				continue
 			}
-			digest := h.Sum(nil)
-			if r.lastAppliedVersion != nil {
-				if bytes.Equal(r.lastAppliedVersion, digest) {
-					// Skip since no meaningful change
-					continue
+
+			for _, configFile := range r.ConfigFiles {
+				configInfo, err := os.Stat(configFile)
+				if err != nil {
+					log.Printf("Error: %s\n", err)
+					continue WaitForEvent
+				}
+				if !os.SameFile(touchedInfo, configInfo) {
+					continue WaitForEvent
```

Comment:
I think here need to just `continue` to be able to check the next cached file:
```suggestion
					continue
```